### PR TITLE
feat: add support for Node 18

### PR DESCRIPTION
**Pull request by cat2608**
#### What does this PR do?

This PR updates Node.js version from 16.16.0 to 18.18.1:

- Updates `.nvmrc` which is used in the Makefile and `create-build-image.sh` script.
- Modifies the engines field in package.json and package-lock.json.
- Adds logs in both the Docker build process and the CircleCI configuration to display the installed Node.js version.

#### Any background context you want to provide?

Previous steps for this job see #4882 #4866

#### What are the relevant tickets?
HEAD-497

- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

### CODE REVIEW

File: .circleci/config.yml
Comment (PeterSchafer) [inline .circleci/config.yml:1]: Issue: The build images (bastiandoetsch209/cli-build) have to be rebuild and updated to use node18 there.

Comment (cat2608) [reply .circleci/config.yml:1]: @PeterSchafer it's [building](https://github.com/snyk/cli/actions/runs/6494341550) 🙌

Comment (PeterSchafer) [reply .circleci/config.yml:1]: Thanks!

Comment (cat2608) [reply .circleci/config.yml:1]: @PeterSchafer the image is ready 🔔

File: .circleci/config.yml [hunk 1]
```diff
@@ -38,12 +38,12 @@ executors:
...
   docker-amd64:
     docker:
-      - image: bastiandoetsch209/cli-build:20230912-103845
+      - image: bastiandoetsch209/cli-build:20231012-101806
     working_directory: /mnt/ramdisk/snyk
     resource_class: large
   docker-arm64:
     docker:
-      - image: bastiandoetsch209/cli-build-arm64:20230912-103845
+      - image: bastiandoetsch209/cli-build-arm64:20231012-101806
```
Comment (cat2608) [inline .circleci/config.yml:46]: @PeterSchafer, updated 💪

**Review by PeterSchafer — APPROVED** [review]

**Review by PeterSchafer — CHANGES_REQUESTED** [review]
Issue: please review the failing tests.

**Pull request closed without merging**